### DISCUSSION TITLE
cloudstack-agent-4.19.1.0  run error  ，Unable to find the guid

### DISCUSSION THREAD

**Discussion by JustUse** [apache/cloudstack#9517] [Q&A]
[root@cloudstack ~]# /usr/bin/java -Djava.io.tmpdir=/usr/share/cloudstack-agent/tmp -Xms256m -Xmx2048m -cp /usr/share/cloudstack-agent/lib/*:/usr/share/cloudstack-agent/plugins/*:/etc/cloudstack/agent:/usr/share/cloudstack-common/scripts com.cloud.agent.AgentShell
log4j:WARN No appenders could be found for logger (com.cloud.agent.AgentShell).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
INFO  [cloud.agent.AgentShell] (main:) (logid:) Agent started
INFO  [cloud.agent.AgentShell] (main:) (logid:) Implementation Version is 4.19.1.0
INFO  [cloud.agent.AgentShell] (main:) (logid:) agent.properties found at /etc/cloudstack/agent/agent.properties
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
Unable to find the guid

[root@cloudstack ~]# cat  /etc/cloudstack/agent/agent.properties | grep -v '#'
guid=
resource=com.cloud.hypervisor.kvm.resource.LibvirtComputingResource
workers=5
host=localhost
port=8250
cluster=default
pod=default
zone=default
local.storage.uuid=
domr.scripts.dir=scripts/network/domr/kvm
hypervisor.type=kvm
iscsi.session.cleanup.enabled=false
[root@cloudstack ~]# 

how can i resolve this problem？thank you very much ！！

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by DaanHoogland** [comment]
@JustUse , the text "Unable to find the guid" occurs only in exceptions that should be logged more extensively to see the stacktrace. Could you edit the log4j.xml file to up the loglevel, please?

Did you start the agent with that command or use the initscript (or add it in cloudstack first)?

**Comment by JustUse** [comment]
> @JustUse , the text "Unable to find the guid" occurs only in exceptions that should be logged more extensively to see the stacktrace. Could you edit the log4j.xml file to up the loglevel, please?
> 
> Did you start the agent with that command or use the initscript (or add it in cloudstack first)?

hi，i found that when  running cloudstack-agent.service and failed in two seconds. is it need to initscript ?  how can i initscript  this?

**Comment by JustUse** [comment]
> @JustUse , the text "Unable to find the guid" occurs only in exceptions that should be logged more extensively to see the stacktrace. Could you edit the log4j.xml file to up the loglevel, please?
> 
> Did you start the agent with that command or use the initscript (or add it in cloudstack first)?

`sed -i 's/INFO/DEBUG/g' /etc/cloudstack/agent/log4j-cloud.xml`

[Turn the agent log level to debug](https://cwiki.apache.org/confluence/display/CLOUDSTACK/KVM+agent+debug
)

[root@cloudstack ~]# /usr/bin/java -Djava.io.tmpdir=/usr/share/cloudstack-agent/tmp -Xms256m -Xmx2048m -cp /usr/share/cloudstack-agent/lib/*:/usr/share/cloudstack-agent/plugins/*:/etc/cloudstack/agent:/usr/share/cloudstack-common/scripts com.cloud.agent.AgentShell
log4j:WARN No appenders could be found for logger (com.cloud.agent.AgentShell).
log4j:WARN Please initialize the log4j system properly.
log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
INFO  [cloud.agent.AgentShell] (main:) (logid:) Agent started
INFO  [cloud.agent.AgentShell] (main:) (logid:) Implementation Version is 4.19.1.0
INFO  [cloud.agent.AgentShell] (main:) (logid:) agent.properties found at /etc/cloudstack/agent/agent.properties
DEBUG [agent.properties.AgentPropertiesFileHandler] (main:) (logid:) Property [host] was altered. Now using the value [localhost].
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
DEBUG [agent.properties.AgentPropertiesFileHandler] (main:) (logid:) Property [guid] has empty or null value. Using default value [null].
DEBUG [agent.properties.AgentPropertiesFileHandler] (main:) (logid:) Property [developer] has empty or null value. Using default value [false].
Unable to find the guid

**Comment by DaanHoogland** [comment]
@JustUse the field 
```
guid=
```
should have a value in the agent properties file. Did you add the host to cloudstack?

I am not sure, but I think  you can make one up and start it.

**Reply by JustUse** [reply]
> @JustUse the field
> 
> ```
> guid=
> ```
> 
> should have a value in the agent properties file. Did you add the host to cloudstack?
> 
> I am not sure, but I think you can make one up and start it.

I just try to deploy cloudstack-agent , just one machine.  but I do not how to set the value for guid  :(

**Reply by JustUse** [reply]
> @JustUse the field
> 
> ```
> guid=
> ```
> 
> should have a value in the agent properties file. Did you add the host to cloudstack?
> 
> I am not sure, but I think you can make one up and start it.

I think it need to add host for making up , but i got error :( 

Could not add host at [http://192.168.124.154] with zone [1], pod [1] and cluster [1] due to: [CloudStack Agent setup through command [cloudstack-setup-agent ] with parameters [ -m 192.168.124.154 -z 1 -p 1 -c 1 -g f51131ca-f9b4-3da9-809c-bdb356365ab7 -a -s --pubNic=cloudbr0 --prvNic=cloudbr0 --guestNic=cloudbr0 --hypervisor=kvm] failed.].

**Reply by DaanHoogland** [reply]
run `uuid` to get a valid value for guid, but that is not needed. You don't need to run the agent by hand. Adding the host to a cluster in cloudstack will take care of that.

first install the host:
https://docs.cloudstack.apache.org/en/latest/installguide/hypervisor/kvm.html#host-kvm-installation
note that this not include the setup, but only the installation of cloudstack-agent

and next add it to you cloudstack installation:
https://docs.cloudstack.apache.org/en/latest/installguide/configuration.html#adding-a-host-xenserver-or-kvm

**Reply by JustUse** [reply]
> @JustUse the field
> 
> ```
> guid=
> ```
> 
> should have a value in the agent properties file. Did you add the host to cloudstack?
> 
> I am not sure, but I think you can make one up and start it.

honestly，I try to resolve this problem for looking cloudstack-management's logs，thank all  thank all : ) ！！！！

**Comment by DaanHoogland** [comment]
closed this for lack of interaction, please reopen or open a new item if applicable
